### ISSUE TITLE
Product edition anormaly slow

### ISSUE THREAD

**Issue by rflorent** [PrestaShop/PrestaShop#15560]
**Describe the bug**
 The edition of a product is very slow when this one have a huge number of price.
In my case, 8 or 10 minutes to wait ! unusable.

**To Reproduce**
Open your web console and trace ajax request (symfony's one is useless, the product legacy controller is still used and, so dumping is not possible)
Edit a product with at least a hundred specific prices and at least 20 features. 
Simply send the request and wait for death.

This is due to the handle of features in the product class.

classes/Product.php (4061)
```
public function addFeaturesToDB($id_feature, $id_value, $cust = 0)
    {
        if ($cust) {
            $row = array('id_feature' => (int) $id_feature, 'custom' => 1);
            Db::getInstance()->insert('feature_value', $row);
            $id_value = Db::getInstance()->Insert_ID();
        }
        $row = array('id_feature' => (int) $id_feature, 'id_product' => (int) $this->id, 'id_feature_value' => (int) $id_value);
        Db::getInstance()->insert('feature_product', $row);
        SpecificPriceRule::applyAllRules(array((int) $this->id));
        if ($id_value) {
            return $id_value;
        }
    }
```
Features can be a condition in a specific price, that's why rules are applied.
In my case with 20 features,  rules are applied 20 times, meaning deleting specific prices and recreate them... crazy

I think rules can be applied only one time, after handling features !

**Additional information**
PrestaShop version: 1.7.5, 1.7.6, all 1.7.*
PHP version: 7.2

**Comment by mariem-abid** [comment]
Hi @rflorent,

Do you use specific modules for the products page?

Thanks!

**Comment by rflorent** [comment]
Hi @mariem-abid 

No specific module.

I've tried a specific module thinking it would speed up the save, the module split the update part by part and send only data concerned (like only features). As it use the same core function, same problem. Btw it pointed me on the right direction, features and specific prices.

I think code needs to be optimized, SpecificPriceRule::applyAllRules(array((int) $this->id)); is called when features are deleted (before and after, why ?) and on each feature added...

I reduce the time to 43 seconds by disable the call on specificPriceRule, and making juste one at the end.  (overrides...)

**Comment by mariem-abid** [comment]
@rflorent, could you please provide us with more info? We need more details to understand how we can reproduce your issue:
 - host
 - debug mode report
 - PHP error logs
 - apache error log
 - javascript console log
 - screenshots

Don't you know how to get this information? Please read the following article:
http://build.prestashop.com/howtos/misc/how-to-create-bug-report/

Thanks!

**Comment by rflorent** [comment]
@mariem-abid hello, this is not a bug, there is no errors, so I can't provide the things asked. Prestashop legacy code core need to be rewrite, as explained, editing a product with a lot of  additionnal data linked (category, specific prices, features) lead to a performance issue.

A product with 20 features, means that rules are applied 20 times, when a rule is applied, specific prices are deleted and created again.  And that just for features...

**Comment by khouloudbelguith** [comment]
Hi @rflorent,

We are working on the product page to fix all issues.
#13891
Also, some issue should be fixed related to the performance: #9809 & https://github.com/PrestaShop/PrestaShop/issues/13698
I close the issue, feel free to open a new one when needed.
Thanks!
